platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationFactory.java

```diff
@@ -275,4 +279,8 @@ DefaultDependencyScopeConfiguration createDependencyScope(
         instance.addMutationValidator(rootComponentMetadataBuilder.getValidator());
         return instance;
     }
+
+    public InternalProblems getProblems() {
```

Comment:
I reworked the `DefaultResolutionHost` a little, but don't want to add a new service into `DefaultConfiguration` as part of this, as it would explode the amount of changes in this PR.  I'm not too concerned with getting the Problems service from the factory instead of keeping a reference in each configuration itself.  

If you think it's a real problem, I can make a issue to follow up on it later.

Reply:
I'd also follow up on this later.

InternalProblems is/will be a widely used interface, there will be some point where it's close to "everywhere"

Reply:
Justin has convinced me to inject this into `DefaultConfiguration` directly via the constructor.  I'm going to do that now, so hopefully that addresses any concerns.